File: agent/sdk/python/examples/http_402_settlement.py
from layerx_sdk import Client, IdempotentMutation, SubmissionExecuted, SubmissionState


def settle_402(client: Client, request: IdempotentMutation[bytes]) -> str:
    state = client.call("submit", request)
    if not isinstance(state, SubmissionExecuted):
        kind = state.kind if isinstance(state, SubmissionState.__args__) else "Invalid"
        raise RuntimeError(f"settlement_failed:{kind}")
    return state.receipt_ref

File: agent/sdk/typescript/examples/http-402-settlement.ts
import { Client, type IdempotentMutation, type SubmissionState } from "../src/index.js";

export interface SettlementRequest { readonly paymentRequirement: Uint8Array }

export async function settle402(
  client: Client,
  request: IdempotentMutation<SettlementRequest>,
): Promise<string> {
  const state = await client.call<typeof request, SubmissionState>("submit", request);
  if (state.kind === "Unknown") throw new Error("settlement_unknown");
  if (state.kind !== "Executed") throw new Error(`settlement_failed:${state.kind}`);
  return state.receiptRef;
}
